### DISCUSSION TITLE
2FA recovery using SSH credentials - Error generating token.

### DISCUSSION THREAD

**Discussion by tccorp** [community/community#120879] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

SSH key 2FA recovery method failing

ssh -T -ai /home/usr/.ssh/github git@github.com verify
Enter passphrase for key '/home/usr/.ssh/github': 
Please provide the following verification token to GitHub Support.
Error generating token.

SSH key is valid:
ssh -i ~/.ssh/github -o IdentityAgent=none -T git@github.com
Enter passphrase for key '/home/usr/.ssh/github': 
Hi usr/project! You've successfully authenticated, but GitHub does not provide shell access.

**Comment by tccorp** [comment]
Debug output:
ssh -vT -ai /home/usrx/.ssh/github [git@github.com](mailto:git@github.com) verify
OpenSSH_8.4p1 Debian-5+deb11u3, OpenSSL 1.1.1w 11 Sep 2023
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files
debug1: /etc/ssh/ssh_config line 21: Applying options for *
debug1: Connecting to github.com [20.205.243.166] port 22.
debug1: Connection established.
debug1: identity file /home/usrx/.ssh/github type 0
debug1: identity file /home/usrx/.ssh/github-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.4p1 Debian-5+deb11u3
debug1: Remote protocol version 2.0, remote software version babeld-9713928c
debug1: no match: babeld-9713928c
debug1: Authenticating to github.com:22 as 'git'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: ecdsa-sha2-nistp256
debug1: kex: server->client cipher: [chacha20-poly1305@openssh.com](mailto:chacha20-poly1305@openssh.com) MAC: compression: none
debug1: kex: client->server cipher: [chacha20-poly1305@openssh.com](mailto:chacha20-poly1305@openssh.com) MAC: compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:p2QAMXNIC1TJYWeIOttrVc98/R1BUFWu3/LiyKgUfQM
debug1: Host 'github.com' is known and matches the ECDSA host key.
debug1: Found key in /home/usrx/.ssh/known_hosts:1
debug1: ssh_packet_send2_wrapped: resetting send seqnr 3
debug1: rekey out after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: ssh_packet_read_poll2: resetting read seqnr 3
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 134217728 blocks
debug1: Will attempt key: /home/usrx/.ssh/githubmur RSA SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx explicit
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<[ssh-ed25519-cert-v01@openssh.com](mailto:ssh-ed25519-cert-v01@openssh.com),[ecdsa-sha2-nistp521-cert-v01@openssh.com](mailto:ecdsa-sha2-nistp521-cert-v01@openssh.com),[ecdsa-sha2-nistp384-cert-v01@openssh.com](mailto:ecdsa-sha2-nistp384-cert-v01@openssh.com),[ecdsa-sha2-nistp256-cert-v01@openssh.com](mailto:ecdsa-sha2-nistp256-cert-v01@openssh.com),[sk-ssh-ed25519-cert-v01@openssh.com](mailto:sk-ssh-ed25519-cert-v01@openssh.com),[sk-ecdsa-sha2-nistp256-cert-v01@openssh.com](mailto:sk-ecdsa-sha2-nistp256-cert-v01@openssh.com),[rsa-sha2-512-cert-v01@openssh.com](mailto:rsa-sha2-512-cert-v01@openssh.com),[rsa-sha2-256-cert-v01@openssh.com](mailto:rsa-sha2-256-cert-v01@openssh.com),[ssh-rsa-cert-v01@openssh.com](mailto:ssh-rsa-cert-v01@openssh.com),[sk-ssh-ed25519@openssh.com](mailto:sk-ssh-ed25519@openssh.com),[sk-ecdsa-sha2-nistp256@openssh.com](mailto:sk-ecdsa-sha2-nistp256@openssh.com),ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey
debug1: Next authentication method: publickey
debug1: Offering public key: /home/usrx/.ssh/github RSA SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx explicit
debug1: Server accepts key: /home/usrx/.ssh/github RSA SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx explicit
Enter passphrase for key '/home/usrx/.ssh/github':
debug1: Authentication succeeded (publickey).
Authenticated to github.com ([20.205.243.166]:22).
debug1: channel 0: new [client-session]
debug1: Entering interactive session.
debug1: pledge: network
debug1: client_input_global_request: rtype [hostkeys-00@openssh.com](mailto:hostkeys-00@openssh.com) want_reply 0
debug1: Sending environment.
debug1: Sending env LC_ADDRESS = en_GB.UTF-8
debug1: Sending env LC_NAME = en_GB.UTF-8
debug1: Sending env LC_MONETARY = en_GB.UTF-8
debug1: Sending env LC_PAPER = en_GB.UTF-8
debug1: Sending env LANG = en_US.UTF-8
debug1: Sending env LC_IDENTIFICATION = en_GB.UTF-8
debug1: Sending env LC_TELEPHONE = en_GB.UTF-8
debug1: Sending env LC_MEASUREMENT = en_GB.UTF-8
debug1: Sending env LC_TIME = en_GB.UTF-8
debug1: Sending env LC_ALL = en_US.UTF-8
debug1: Sending env LC_NUMERIC = en_GB.UTF-8
debug1: Sending command: verify
Please provide the following verification token to GitHub Support.debug1: client_input_channel_req: channel 0 rtype exit-status reply 0

Error generating token.
debug1: channel 0: free: client-session, nchannels 1
Transferred: sent 4148, received 3120 bytes, in 0.6 seconds
Bytes per second: sent 7512.7, received 5650.8
debug1: Exit status 1

It seems that today, bold text appeared, whereas it wasn't displayed in previous attempts.

**Comment by gil-son** [comment]
Hello, how are you? I feel for that situation, you need to verify it

Try to reconfigure your 2FA. When you created it, you received some sequence of words to recovery your access whether necessary

<b>Or</b>

Access the [Github Support](https://support.github.com/)

Click on My Ticket -> New Ticket -> Finally describe it

If I supported you consider to mark my answer. Good Luck!

**Reply by tccorp** [reply]
Thanks, but it is impossible to reconfigure 2FA without login, so my last option is ssh keys, which is not working, i have no backup codes and verified device. Github support keep silence for two weeks already.

**Reply by gil-son** [reply]
I understand

In this case, consider to verify with Github Support. They can check it and provide some alternatives for you

**Reply by tccorp** [reply]
Hope they will answer during this year, but since my keys is valid and authentication with it is ok, it is a bug in the github ssh subsystem. You can help by upvoting the first message to escalate attention to this matter.
